src/const_eval.md

```diff
@@ -42,7 +42,10 @@ r[const-eval.const-expr.path-item]
   Recursively defining constants is not allowed.
 
 r[const-eval.const-expr.path-static]
-* Paths to [statics]. These are only allowed within the initializer of a static.
+* Paths to [statics] with these restrictions and observations.
+  * Reads and writes to `extern` statics are not allowed in any constant evaluation context.
+  * If the evaluation is carried out in an initializer of a `static` item, then *writes* into mutable `static` are not allowed, such as `static mut`s and `static`s containing an interior-mutable type.
+  * If the evaluation is *not* carried out in an initializer of a `static` item, then reads and writes to any mutable `static`, such as `static mut`s and `static`s containing an interior-mutable type.
```

Comment:
Something (a verb?) seems to have been lost in this sentence. Maybe this?

```suggestion
  * If the evaluation is *not* carried out in an initializer of a `static` item, then reads and writes to any mutable `static`, such as `static mut`s and `static`s containing an interior-mutable type, are not allowed.
```

Reply:
With writes being handled separately by my suggestion above, this can just say "reads from any mutable `static".